site/content/extend/plugins/community-plugin-marketplace.md

```diff
@@ -0,0 +1,79 @@
+---
+title: Community Plugins in the Marketplace
+subsection: Plugins (Beta)
+weight: 60
+---
+
+Once your plugins has reached a certain level of quality, you might consider submitting it to the Plugin Marketplace. The Plugin Marketplace allows discovery, installation and updates of plugins directly within Mattermost. It is a great way to get feedback on your plugin and help making it more popular. Once your plugin is accepted to the Marketplace, Mattermost will also send you swag!
+
+### Requirements for adding community plugin to the Marketplace
+Every community plugin must fulfill the following checklist to be added to the Marketplace:
+
+**Product Requirements (Checked by a Product Manager)**
+
+1. The plugin is published under an Apache v2 compatible license (e.g. no GPL, APGL). A list of compatible licenses can be found [here](https://apache.org/legal/resolved.html#category-a).
+2. The source code is available in a public git repository.
+3. There is a public issue or bug tracker for the plugin, which is linked in the plugin documentation and linked via `support_url` in the manifest.
+4. The plugin provides detailed usage documentation with at least one screenshot of the plugin in action, list of features and a development guide. This is typically a README file or a landing page on the web. The link to the documentation is set as `homepage_url` in the manifest. A great example is the [README of the GitHub plugin](https://github.com/mattermost/mattermost-plugin-github/blob/master/README.md).
+5. For every release a changelog has to be publish. The link to it is noted via `release_notes_url` in the manifest.
+6. The plugin has to be out of Beta and be released with at least v1.0.0.
```

Comment:
5-7 is not 100% (yet) true for ourselves, why are we imposing it on the community plugins?

Reply:
I saw the beta section at the bottom, maybe link there from here?

Reply:
> 5-7 is not 100% (yet) true for ourselves, why are we imposing it on the community plugins?

Mostly to keep increasing the quality of plugins in the Marketplace. Especially 7 is a huge pain point for admins.

Reply:
> I saw the beta section at the bottom, maybe link there from here?

Do you mean linking from the beta section to this requirement?

Reply:
7) is going to force devs to use the KV store and proprietary config methods more. 1/5 we should offer best practices of configuration management with a CLI interface, and JSON export/import (via CLI), and continue encouraging to use the config store (and eventually the admin console?) for seldom changing values.

I am ok with including it as a requirement, as long as we provide at least a somewhat specific future direction for what may be coming, even if a link to a "Needs Spec" ticket to contribute use-cases to.

Reply:
@marianunez recently implemented the possibility to have custom settings in the system console using js. See https://github.com/mattermost/mattermost-plugin-custom-attributes/pull/18 for an example usage. That seams to me a very good first step on helping plugin defining everything in the system console.